modules/cudacodec/src/video_writer.cpp

```diff
@@ -61,12 +61,6 @@ GUID EncodingProfileGuid(const EncodeProfile encodingProfile);
 GUID EncodingPresetGuid(const EncodePreset nvPreset);
 bool Equal(const GUID& g1, const GUID& g2);
 
-EncoderParams::EncoderParams() : nvPreset(ENC_PRESET_P3), tuningInfo(ENC_TUNING_INFO_HIGH_QUALITY), encodingProfile(ENC_CODEC_PROFILE_AUTOSELECT),
```

Comment:
There are should be implementation in the `#else` branch:

```
#else // !defined HAVE_NVCUVENC

... CV_Error(...NotImplemented...) implementation stubs ...
```

Reply:
`createVideoWriter` has this else branch. With the changes parameters structure could be created correctly without linker issue, but `createVideoWriter` raises the issue. I propose to merge the PR as is.

Reply:
I moved it to the header to mirror the way [cudacodec::FormatInfo()](https://github.com/opencv/opencv_contrib/blob/4.x/modules/cudacodec/include/opencv2/cudacodec.hpp#L304) is implemented.

Both structs can exist without the Nvidia Video Coded SDK being installed whether they should or not is a different matter.  Do you want me to move the implementation of both into the .cpp and prevent their use when `HAVE_NVCUVENC` and/or `HAVE_NVCUVID` are not defined?

Reply:
I prefer to have config structures always available how it's done in PR.